### ISSUE TITLE
Add blue light detection for theme switch

### ISSUE THREAD

**Issue by thebigsmileXD** [AutoDarkMode/Windows-Auto-Night-Mode#256]
http://stackoverflow.com/questions/43340619/ddg#43953978

It would be cool to have a mode to detect if the blue light filter is running, as alternative to the time based / location based mode

**Comment by Armin2208** [comment]
oh wow! thanks.
But it would be probably hard to implement. :D

**Comment by Spiritreader** [comment]
> oh wow! thanks.
> But it would be probably hard to implement. :D

It is potentially possible using WMI registry monitoring if we investigate this further.

However something very important to note:
If this can really be used to detect night/day dependind on Windows Night Light, it would require a completely new mode that runs without timer, because the underlying theme switch trigger is now an event!

**Comment by thebigsmileXD** [comment]
Working on it on a fork https://github.com/thebigsmileXD/Windows-Auto-Night-Mode/

**Comment by Spiritreader** [comment]
Just a fair bit of info for you:

Armin was very correct in estimating this to be a lot of work if done right.

This cannot be implemented using a timer, because this would mean that there's a check for a registry key update every 60 seconds, which is something we chose to avoid in ADM.
States are only tracked in memory and synchronized once.

We need a new switch source that is not the timer. 
So essentially, what is necessary is to use a change event listener on the regkey. 
As far as I'm aware, this event handler needs to use WMIC queries.
When the TimedSwitch module is disabled, the offset feature would need to reimplemented to work with that. 
For this, a new mechanism has to be created that is able to correctly postpone a theme switch if it was triggered by windows night light. 

Once that is done, postpone compatibility has to be ensured with all modules, for example the GPU monitoring module.

Finally, there has to be some sort of management class which is able to switch between sources (timer based, event based without restarting the service.

All while doing that on the very dev-active beta 5 branch which currently undergoes heavy restructuring 

There's a good chance I forgot some stuff, but yeah, a quite ambitious project.

**Comment by Spiritreader** [comment]
The Auto Dark Mode Version X structure is now final and has been merged into master.
This version will release on Oct 5th.

If you're still interested in getting this implemented in a future release by doing it "the right way", feel free to reach out.
I'll gladly provide more details explanation on how the inner structure works.

**Comment by Spiritreader** [comment]
Is there still interest for this feature @thebigsmileXD (or anyone else who wants this as a matter of fact) 
If so, we now have everything ready to get this working.

**Comment by Spiritreader** [comment]
Closing in favor of #489, thanks again for the contribution, helped a lot in getting this to work!
